test/speech/index.js

```diff
@@ -0,0 +1,836 @@
+        callback(null, FILES);
+      };
+    });
+
+    after(function() {
+      Speech.findFiles_ = findFiles_;
+    });
+
+    it('should find the files', function(done) {
+      Speech.findFiles_ = function(files) {
+        assert.strictEqual(files, FILE);
+        done();
+      };
+
+      speech.detect(FILE, REQ, assert.ifError);
+    });
+
+    it('should return an error from findFiles_', function(done) {
+      var error = new Error('Error.');
+
+      Speech.findFiles_ = function(files, callback) {
+        assert.strictEqual(files, FILE);
+        callback(error);
+      };
+
+      speech.detect(FILE, REQ, function(err) {
+        assert.strictEqual(err, error);
+        done();
+      });
+    });
+
+    it('should return empty detections when none were found', function(done) {
+      speech.recognize = function(REQ, callback) {
+        callback(null, new speech.protos.Speech.SyncRecognizeResponse({
+          results: []
+        }));
+      };
+
+      speech.detect(
+        FILE,
+        REQ,
+        function(err, response) {
+          assert.ifError(err);
+          assert.deepEqual(response, []);
+          done();
+        }
+      );
+    });
+
+    it('should return empty detections when result is empty', function(done) {
+      speech.recognize = function(REQ, callback) {
+        callback(null, new speech.protos.Speech.SyncRecognizeResponse({
+          results: [
+            {
+              alternatives: []
+            }
+          ]
+        }));
+      };
+
+      speech.detect(
+        FILE,
+        REQ,
+        function(err, response) {
+          assert.ifError(err);
+          assert.deepEqual(response, []);
+          done();
+        }
+      );
+    });
+
+    it('should return the correct detections', function(done) {
+      var expected = {
+        transcript: 'foo',
+        confidence: 1.00
+      };
+      var RES = new speech.protos.Speech.SyncRecognizeResponse({
+        results: [
+          {
+            alternatives: [
+              expected
+            ]
+          }
+        ]
+      });
+
+      speech.recognize = function(config, callback) {
+        callback(null, RES);
+      };
+
+      speech.detect(
+        FILE,
+        REQ,
+        function(err, response) {
+          assert.ifError(err);
+          assert.deepEqual(response, [expected]);
+
+          done();
+        }
+      );
+    });
+
+    it('should return an error from recognize()', function(done) {
+      var error = new Error('Error.');
+      var RES = {};
+
+      speech.recognize = function(config, callback) {
+        callback(error, null, RES);
+      };
+
+      speech.detect(FILE, REQ, function(err, response, apiResponse) {
+        assert.strictEqual(err, error);
+        assert.strictEqual(response, null);
+        assert.strictEqual(apiResponse, RES);
+        done();
+      });
+    });
+
+    it('should return the resp from recognize()', function(done) {
```

Comment:
I think we can safely remove this test if we were to pass in `apiResponse` in earlier tests.